zinc/src/main/scala/sbt/internal/inc/MixedAnalyzingCompiler.scala

```diff
@@ -143,10 +143,11 @@ final class MixedAnalyzingCompiler(
     val isPickleJava = config.currentSetup.order == Mixed && config.incOptions.pipelining && javaSrcs.nonEmpty
 
     val earlyOut = config.earlyOutput.flatMap(_.getSingleOutputAsPath.toOption)
-    earlyOut foreach { out =>
-      if (out.toString.endsWith(".jar") && !Files.exists(out)) {
+    val pickleWrite = earlyOut.toList.flatMap { out =>
+      val sbv = scalac.scalaInstance.version.take(4)
+      if (out.toString.endsWith(".jar") && !Files.exists(out))
         scala.reflect.io.RootPath(out, writable = true).close() // creates an empty jar
-      }
+      List("-Ypickle-write", out.toString).filter(_ => sbv == "2.12" || sbv == "2.13")
```

Comment:
In general Zinc, should avoid adding Scalac options since the build user wouldn't be able to control it.
`-Y` flag especially has no guarantee and it's not even supported in all of 2.12 or 2.13.

Reply:
Do you want to send a PR with a way to address this, without breaking the initial use case?

Reply:
What is the initial use case?